main.tsx

```diff
@@ -175,27 +190,62 @@ async (req: Request, info: Deno.ServeHandlerInfo) => {
 
 const graphql_handler = (
     args: {
-        password: string;
+        password?: string;
         kv: Deno.Kv;
         policyStore: PolicyStore;
         relayInformationStore: RelayInformationStore;
     },
 ) =>
 async (req: Request) => {
-    const { password, policyStore } = args;
     if (req.method == "POST") {
-        const query = await req.json();
-        const pw = req.headers.get("password");
-        if (pw != password) {
-            return new Response(`{"errors":"incorrect password"}`);
+        try {
+            const query = await req.json();
+            if (!args.password) {
+                const cookie = req.headers.get("cookie");
+                const token = cookie?.split(";").find((c) => c.includes("token"))?.split("=")[1].split(
+                    " ",
+                )[1];
```

Comment:
This line is not readable at all.
See https://deno.land/std@0.148.0/http/mod.ts?s=getCookies